tests/rptest/tests/end_to_end.py

```diff
@@ -177,10 +177,12 @@ def has_finished_consuming():
                     return False
             return True
 
-        wait_until(has_finished_consuming,
-                   timeout_sec=timeout_sec,
-                   err_msg="Consumer failed to consume up to offsets %s after waiting %ds, last consumed offsets: %s." %\
-                   (str(last_acked_offsets), timeout_sec, list(self.last_consumed_offsets)))
+        wait_until(
+            has_finished_consuming,
+            timeout_sec=timeout_sec,
+            err_msg=
+            f"Consumer failed to consume up to offsets {str(last_acked_offsets)} after waiting {timeout_sec}s, last committed offsets: {self.consumer.get_committed_offsets()}."
```

Comment:
the `err_msg` is formatted and the value of `get_committed_offsets()` is taken before we try `has_finished_consuming` even once, I think that may be misleading

Reply:
you are right, it is a snapshot, need to change this 🤦🏼